### PR TITLE
[refactor] Readme

### PR DESCRIPTION
**Pull request by rennokki**
Managed to re-write a bunch of the configuration and re-thinking the examples a bit cuz some didn't make sense.

You can see the README here: https://github.com/rennokki/laravel-mongodb/tree/refactor/readme

Tree

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -119,301 +156,238 @@ docker-compose up
...
+    'host' => env('DB_HOST', '127.0.0.1'),
+    'port' => env('DB_PORT', 27017),
+    'database' => env('DB_DATABASE', 'homestead'),
+    'username' => env('DB_USERNAME', 'homestead'),
+    'password' => env('DB_PASSWORD', 'secret'),
+    'options' => [
+        'database' => 'admin', // required with Mongo 3+
+
+        // here you can pass more settings
+        // see https://www.php.net/manual/en/mongoclient.construct.php under "Parameters" for a list of complete parameters you can use
```
Comment (Smolevich) [inline README.md]: Link on deprecated extension

Comment (Smolevich) [reply README.md]: Actual link [here](https://www.php.net/manual/en/mongodb-driver-manager.construct.php)

File: README.md [hunk 2]
```diff
@@ -119,301 +156,238 @@ docker-compose up
...
 
-This will allow you to use the registered alias like:
+class Book extends Model
+{
+    protected $primaryKey = 'id';
+}
 
-```php
-class MyModel extends Moloquent {}
+// Mongo will also createa _id, but the 'id' property will be used for primary key actions like find().
```
Comment (Smolevich) [inline README.md]:
```suggestion
// Mongo will also create _id, but the 'id' property will be used for primary key actions like find().
```

File: README.md (hunk 1 above)
Comment (rennokki) [reply README.md]: Can you change it to https://www.php.net/manual/en/mongodb-driver-manager.construct.php ?

File: README.md [hunk 3]
```diff
@@ -911,178 +927,191 @@ You can replace the embedded model with a new model like this:
...
+Last, add the service provider in `config/app.php`:
 
 ```php
-DB::connection()->disableQueryLog();
+Jenssegers\Mongodb\MongodbQueueServiceProvider::class,
 ```
 
-*From: https://laravel.com/docs/4.2/database#query-logging*
+### Sentry
+If you want to use this library with [Sentry](https://cartalyst.com/manual/sentry), then check out https://github.com/jenssegers/Laravel-MongoDB-Sentry
```
Comment (Smolevich) [inline README.md]: What is the purpose to add link for deprecated package?

File: README.md [hunk 4]
```diff
@@ -3,33 +3,52 @@ Laravel MongoDB
...
-* [Eloquent](#eloquent)
-* [Optional: Alias](#optional-alias)
-* [Query Builder](#query-builder)
-* [Schema](#schema)
-* [Extensions](#extensions)
-* [Examples](#examples)
-
-Installation
+- [Laravel MongoDB](#laravel-mongodb)
+  - [Table of contents](#table-of-contents)
```
Comment (Smolevich) [inline README.md]: I think that this paragraph is redundant

**Comment by rennokki** [comment]
@Smolevich I don't think there are more grammar errors. I used Grammarly and commited here https://github.com/jenssegers/laravel-mongodb/pull/1917/commits/7e91b7a00c8fd34695a14f83abcd3733c1d1bdd6

**Comment by Smolevich** [comment]
@jenssegers please see on PR, are you agree with reformatting?

**Comment by divine** [comment]
Well, if are going to refactor readme, we should take care for this PR https://github.com/jenssegers/laravel-mongodb/pull/1121 too.

**Comment by divine** [comment]
Also we need take care for this too https://github.com/jenssegers/laravel-mongodb/pull/1830

File: README.md [hunk 5]
```diff
@@ -3,33 +3,48 @@ Laravel MongoDB
...
-* [Configuration](#configuration)
-* [Eloquent](#eloquent)
-* [Optional: Alias](#optional-alias)
-* [Query Builder](#query-builder)
-* [Schema](#schema)
-* [Extensions](#extensions)
-* [Examples](#examples)
-
-Installation
+Laravel Eloquent adds support for ODM (Object Document Mapper) to Laravel. It's the same as Eloquent ORM, but with Documents, since MongoDB is a NoSQL database.
```
Comment (jenssegers) [inline README.md]: Not sure this intro makes it easier to understand what this package does. The main point is to basically add support for a mongodb with the ability to use all of the eloquent and query builder goodies Laravel has.

File: README.md [hunk 6]
```diff
@@ -78,34 +106,39 @@ Upgrading
 
 #### Upgrading from version 2 to 3
```
Comment (jenssegers) [inline README.md]: Should we remove this part? Not sure if version 2 is still being used a lot.

**Comment by jenssegers** [comment]
I like the structure. Probably a lot easier to find documentation with this new index.

File: README.md [hunk 7]
```diff
@@ -3,33 +3,48 @@ Laravel MongoDB
...
-* [Query Builder](#query-builder)
-* [Schema](#schema)
-* [Extensions](#extensions)
-* [Examples](#examples)
-
-Installation
+Laravel Eloquent adds support for ODM (Object Document Mapper) to Laravel. It's the same as Eloquent ORM, but with Documents, since MongoDB is a NoSQL database.
+
+
+- [Laravel MongoDB](#laravel-mongodb)
```
Comment (jenssegers) [inline README.md]: I think we can skip this first level to make it look nicer.

Comment (rennokki) [reply README.md]: This is what VSCode adds because it finds this at the top:

```markdown
Laravel MongoDB
===============
```

File: README.md [hunk 8]
```diff
@@ -119,301 +111,238 @@ docker-compose up
...
-        'database' => 'admin' // sets the authentication database required by mongo 3
-    ]
+    'driver' => 'mongodb',
+    'host' => env('DB_HOST', '127.0.0.1'),
+    'port' => env('DB_PORT', 27017),
+    'database' => env('DB_DATABASE', 'homestead'),
+    'username' => env('DB_USERNAME', 'homestead'),
+    'password' => env('DB_PASSWORD', 'secret'),
+    'options' => [
+        'database' => 'admin', // required with Mongo 3+
```
Comment (Giacomo92) [inline README.md]: For some providers can be different from admin. 
For example using Heroku auth admin is equal to env('DB_DATABASE')

File: README.md [hunk 9]
```diff
@@ -911,178 +882,231 @@ You can replace the embedded model with a new model like this:
...
 ```
 
-If you don't want duplicate items, set the third parameter to `true`:
+If you want to use MongoDB to handle failed jobs, change the database in `config/queue.php`:
 
 ```php
-DB::collection('users')->where('name', 'John')->push('items', 'boots', true);
+'failed' => [
+    'driver' => env('QUEUE_FAILED_DRIVER', 'database'),
+    'database' => env('DB_CONNECTION', 'mongodb'),
```
Comment (Giacomo92) [inline README.md]: 'database' => env('DB_DATABASE, 'homestead'),

File: README.md [hunk 10]
```diff
@@ -119,301 +111,238 @@ docker-compose up
...
 
-And add the service provider in `config/app.php`:
+**Where**
 
 ```php
-Jenssegers\Mongodb\MongodbQueueServiceProvider::class,
+$users =
+    User::where('age', '>', 18)
+        ->take(10)
+        ->get();
```
Comment (Giacomo92) [inline README.md]: I would also add where on embedded models:
`$posts =
    Post::where('category.name', 'sport')
        ->get();`

File: README.md (hunk 8 above)
Comment (rennokki) [reply README.md]: On some projects i use it as

```php
'database' => env('DB_AUTHENTICATION_DATABASE', 'admin'),
```

Do this sound good?

Comment (Giacomo92) [reply README.md]: Perfect!

File: README.md (hunk 9 above)
Comment (rennokki) [reply README.md]: Check this: https://github.com/laravel/laravel/blob/master/config/queue.php#L84

**Comment by rennokki** [comment]
This is how it's looking so far:

Screenshot 2020-01-23 at 12 30 18

File: README.md [hunk 11]
```diff
@@ -915,178 +886,231 @@ You can replace the embedded model with a new model like this:
...
+],
 ```
 
-If you don't want duplicate items, set the third parameter to `true`:
+If you want to use MongoDB to handle failed jobs, change the database in `config/queue.php`:
 
 ```php
-DB::collection('users')->where('name', 'John')->push('items', 'boots', true);
+'failed' => [
+    'driver' => env('QUEUE_FAILED_DRIVER', 'database'),
```
Comment (Smolevich) [inline README.md]: What is the profit to determine new env variable  `QUEUE_FAILED_DRIVER` instead `QUEUE_DRIVER`

Comment (Smolevich) [reply README.md]: Ok, you copy from laravel implementation https://github.com/laravel/laravel/blob/master/config/queue.php#L84

**Comment by rennokki** [comment]
@Giacomo92 & @Smolevich 

Final reviews?

File: README.md [hunk 12]
```diff
@@ -126,10 +126,10 @@ You can use MongoDB either as the main database, either as a side database. To d
...
     'password' => env('DB_PASSWORD', 'secret'),
     'options' => [
-        'database' => 'admin', // required with Mongo 3+
-
-        // here you can pass more settings
-        // see https://www.php.net/manual/en/mongodb-driver-manager.construct.php under "Uri Options" for a list of complete parameters you can use
+        // here you can pass more settings to the Mongo Driver Manager
+        // see https://www.php.net/manual/en/mongodb-driver-manager.construct.php under "Uri Options" for a list of complete parameters that you can use
+        
+        'authSource' => env('DB_AUTHENTICATION_DATABASE', 'admin'), // required with Mongo 3+
```
Comment (Giacomo92) [inline README.md]: I never user `'authSource'`, I have always used `'database'.  Is this correct?

I am also using 
`'dump' => [
		        'mongodb_user_auth' => env('DB_AUTHENTICATION_DATABASE', 'admin') //Auth for laravel backup plugin
	        ]` 
for backup. But I do not know if it is a specific configuration of [spatie backup](https://docs.spatie.be/laravel-backup/v6/installation-and-setup/#dumping-the-database) and if it makes sense to mention it.

Comment (rennokki) [reply README.md]: I had the weird feeling that `database` is not quite what i was using before and somehow had to look up for the available settings.

The setting for the dump can be added by the users that use Spatie's Backup package.

Comment (Giacomo92) [reply README.md]: Ok for the dump option. It is ok for me. You can merge

**Pull request merged**